backend/app/services/site_analyzer.py

```
@@ -0,0 +1,288 @@
+import httpx
+from bs4 import BeautifulSoup
+from typing import Dict, List, Optional, Tuple
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)